### ISSUE TITLE
64-bit getcwd error due to rounding of device number

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5429]
Migrated from [rt.perl.org#9220](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=9220) (status was 'resolved')

Searchable as RT9220$

**Comment by p5pRT** [comment]
### From k.tomlinson@auckland.ac.nz

#### Created by k.tomlinson@auckland.ac.nz


This is a bug report for perl from k\.tomlinson@&#8203;auckland\.ac\.nz\,
generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

File&#8203;::Find\.pm and FindBin\.pm fail their tests when building 64bitall
due to an error in Cwd&#8203;:getcwd\.  Cwd&#8203;:abs\_path compares device numbers
\(which have the most significant 64th bit set on AIX\) and doesn't
detect a difference in the less significant bits\.  This results in an
invalid directory from getcwd&#8203;:

% 32/perl \-we 'BEGIN\{@&#8203;INC="32/lib"\}; use Cwd; print getcwd\(\)\,"\\n"'
/home/soe/tomlinso/perl
% 64/perl \-we 'BEGIN\{@&#8203;INC="64/lib"\}; use Cwd; print getcwd\(\)\,"\\n"'
/soe/tomlinso/perl

The incorrect directory is a result of abs\_path believing that /home
and / are the same directory&#8203;:

% 64/perl \-we 'printf "%x %x\\n"\, \(stat "/home"\)\[0\.\.1\]'
8000000a00000008 2
% 64/perl \-we 'printf "%x %x\\n"\, \(stat "/"\)\[0\.\.1\]'
8000000a00000004 2
% 32/perl \-we 'printf "%x %x\\n"\, \(stat "/home"\)\[0\.\.1\]'
a0008 2
% 32/perl \-we 'printf "%x %x\\n"\, \(stat "/"\)\[0\.\.1\]'
a0004 2

Simply including

use integer

in Cwd\.pm corrects this problem\.  I assume the device numbers were being
converted to doubles before comparison\.

This works fine for 32 and 64\-bit perls on AIX\, but I don't know
whether this could cause problems for 32\-bit perls on a system with
64\-bit device numbers\.  Would the \`use integer' truncate the device
number to 32\-bits?  Does that mean that using \`eq' and \`ne' instead of
\`==' and \`\!=' is a better solution?

<details><summary>Perl Info</summary>

``` ---
Flags:
    category=library
    severity=high

Site configuration information for perl v5.6.1:

Configured by tomlinso at Wed May  8 17:26:18 NZST 2002.

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=aix, osvers=5.1.0.0, archname=aix-64all
    uname='aix hpc1 1 5 0020abea4c00 '
    config_args='-de -Dmksymlinks -Dprefix=/product/cmiss -Dbin=/product/cmiss/bin/64 -Dsitebin=/product/cmiss/bin/64 -Duse64bitall -Ausenm=false'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=define use64bitall=define uselongdouble=undef
  Compiler:
    cc='cc', ccflags ='-D_ALL_SOURCE -D_ANSI_C_SOURCE -D_POSIX_SOURCE -qmaxmem=16384 -I/usr/local/include  -D_LARGE_FILES -qlonglong -q64 -DUSE_64_BIT_ALL',
    optimize='-O',
    cppflags='-D_ALL_SOURCE -D_ANSI_C_SOURCE -D_POSIX_SOURCE -qmaxmem=16384 -I/usr/local/include'
    ccversion='undef', gccversion='', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=87654321
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='ld', ldflags ='-L/usr/local/lib  -q64'
    libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib
    libs=-lbind -lnsl -ldbm -ldl -lld -lm -lC -lc -lcrypt -lbsd -lPW -liconv
    perllibs=-lbind -lnsl -ldl -lld -lm -lC -lc -lcrypt -lbsd -lPW -liconv
    libc=, so=a, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_aix.xs, dlext=so, d_dlsymun=undef, ccdlflags='  -bE:/product/cmiss/lib/perl5/5.6.1/aix-64all/CORE/perl.exp'
    cccdlflags=' ', lddlflags='-b64 -bhalt:4 -bM:SRE -bI:$(PERL_INC)/perl.exp -bE:$(BASEEXT).exp -b noentry -lC -lc -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.1:
    /product/cmiss/lib/perl5/5.6.1/aix-64all
    /product/cmiss/lib/perl5/5.6.1
    /product/cmiss/lib/perl5/site_perl/5.6.1/aix-64all
    /product/cmiss/lib/perl5/site_perl/5.6.1
    /product/cmiss/lib/perl5/site_perl
    .


Environment for perl v5.6.1:
    HOME=/home/soe/tomlinso
    LANG=en_US
    LANGUAGE (unset)
    LC__FASTMSG=true
    LD_LIBRARY_PATH (unset)
    LIBPATH (unset)
    LOGDIR (unset)
    PATH=/home/soe/tomlinso/bin:/product/cmiss/bin:/usr/local/bin:/opt/freeware/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/vacpp/bin:/usr/bin/X11:/sbin:/usr/java130/jre/bin:/usr/java130/bin
    PERL_BADLANG (unset)
    SHELL=/product/cmiss/bin/zsh


```
</details>

**Comment by p5pRT** [comment]
### From @jhi

On Wed\, May 08\, 2002 at 06&#8203;:44&#8203;:06PM \+1200\, Karl Tomlinson wrote&#8203;:
> 
> This is a bug report for perl from k\.tomlinson@&#8203;auckland\.ac\.nz\,
> generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> 
> File&#8203;::Find\.pm and FindBin\.pm fail their tests when building 64bitall
> due to an error in Cwd&#8203;:getcwd\.  Cwd&#8203;:abs\_path compares device numbers
> \(which have the most significant 64th bit set on AIX\) and doesn't
> detect a difference in the less significant bits\.  This results in an
> invalid directory from getcwd&#8203;:

Could you try the latest 5\.8\.0\-to\-be developer snapshot&#8203;:

&nbsp;        http&#8203;://www\.iki\.fi/jhi/perl@&#8203;16493\.tgz
&nbsp;        http&#8203;://www\.iki\.fi/jhi/perl@&#8203;16493\.tbz
&nbsp;  
&nbsp;        ftp&#8203;://ftp\.funet\.fi/pub/languages/perl/snap/perl@&#8203;16493\.tgz
&nbsp;        ftp&#8203;://ftp\.funet\.fi/pub/languages/perl/snap/perl@&#8203;16493\.tbz
&nbsp;      
&nbsp;        ftp\.funet\.fi&#8203;::perlsnap

it has several fixes related to 64\-bitness\.

> % 32/perl \-we 'BEGIN\{@&#8203;INC="32/lib"\}; use Cwd; print getcwd\(\)\,"\\n"'
> /home/soe/tomlinso/perl
> % 64/perl \-we 'BEGIN\{@&#8203;INC="64/lib"\}; use Cwd; print getcwd\(\)\,"\\n"'
> /soe/tomlinso/perl
> 
> The incorrect directory is a result of abs\_path believing that /home
> and / are the same directory&#8203;:
> 
> % 64/perl \-we 'printf "%x %x\\n"\, \(stat "/home"\)\[0\.\.1\]'
> 8000000a00000008 2
> % 64/perl \-we 'printf "%x %x\\n"\, \(stat "/"\)\[0\.\.1\]'
> 8000000a00000004 2
> % 32/perl \-we 'printf "%x %x\\n"\, \(stat "/home"\)\[0\.\.1\]'
> a0008 2
> % 32/perl \-we 'printf "%x %x\\n"\, \(stat "/"\)\[0\.\.1\]'
> a0004 2
> 
> Simply including
> 
> use integer
> 
> in Cwd\.pm corrects this problem\.  I assume the device numbers were being
> converted to doubles before comparison\.
> 
> This works fine for 32 and 64\-bit perls on AIX\, but I don't know
> whether this could cause problems for 32\-bit perls on a system with
> 64\-bit device numbers\.  Would the \`use integer' truncate the device
> number to 32\-bits?  Does that mean that using \`eq' and \`ne' instead of
> \`==' and \`\!=' is a better solution?
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\- \-\-\-
> Flags&#8203;:
>     category=library
>     severity=high
> \-\-\-
> Site configuration information for perl v5\.6\.1&#8203;:
> 
> Configured by tomlinso at Wed May  8 17&#8203;:26&#8203;:18 NZST 2002\.
> 
> Summary of my perl5 \(revision 5\.0 version 6 subversion 1\) configuration&#8203;:
>   Platform&#8203;:
>     osname=aix\, osvers=5\.1\.0\.0\, archname=aix\-64all
>     uname='aix hpc1 1 5 0020abea4c00 '
>     config\_args='\-de \-Dmksymlinks \-Dprefix=/product/cmiss \-Dbin=/product/cmiss/bin/64 \-Dsitebin=/product/cmiss/bin/64 \-Duse64bitall \-Ausenm=false'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
>     useperlio=undef d\_sfio=undef uselargefiles=define usesocks=undef
>     use64bitint=define use64bitall=define uselongdouble=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-D\_ALL\_SOURCE \-D\_ANSI\_C\_SOURCE \-D\_POSIX\_SOURCE \-qmaxmem=16384 \-I/usr/local/include  \-D\_LARGE\_FILES \-qlonglong \-q64 \-DUSE\_64\_BIT\_ALL'\,
>     optimize='\-O'\,
>     cppflags='\-D\_ALL\_SOURCE \-D\_ANSI\_C\_SOURCE \-D\_POSIX\_SOURCE \-qmaxmem=16384 \-I/usr/local/include'
>     ccversion='undef'\, gccversion=''\, gccosandvers=''
>     intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=87654321
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=8
>     ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
>     alignbytes=8\, usemymalloc=n\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='ld'\, ldflags ='\-L/usr/local/lib  \-q64'
>     libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib
>     libs=\-lbind \-lnsl \-ldbm \-ldl \-lld \-lm \-lC \-lc \-lcrypt \-lbsd \-lPW \-liconv
>     perllibs=\-lbind \-lnsl \-ldl \-lld \-lm \-lC \-lc \-lcrypt \-lbsd \-lPW \-liconv
>     libc=\, so=a\, useshrplib=false\, libperl=libperl\.a
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_aix\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='  \-bE&#8203;:/product/cmiss/lib/perl5/5\.6\.1/aix\-64all/CORE/perl\.exp'
>     cccdlflags=' '\, lddlflags='\-b64 \-bhalt&#8203;:4 \-bM&#8203;:SRE \-bI&#8203;:$\(PERL\_INC\)/perl\.exp \-bE&#8203;:$\(BASEEXT\)\.exp \-b noentry \-lC \-lc \-L/usr/local/lib'
> 
> Locally applied patches&#8203;:
>     
> 
> \-\-\-
> @&#8203;INC for perl v5\.6\.1&#8203;:
>     /product/cmiss/lib/perl5/5\.6\.1/aix\-64all
>     /product/cmiss/lib/perl5/5\.6\.1
>     /product/cmiss/lib/perl5/site\_perl/5\.6\.1/aix\-64all
>     /product/cmiss/lib/perl5/site\_perl/5\.6\.1
>     /product/cmiss/lib/perl5/site\_perl
>     \.
> 
> \-\-\-
> Environment for perl v5\.6\.1&#8203;:
>     HOME=/home/soe/tomlinso
>     LANG=en\_US
>     LANGUAGE \(unset\)
>     LC\_\_FASTMSG=true
>     LD\_LIBRARY\_PATH \(unset\)
>     LIBPATH \(unset\)
>     LOGDIR \(unset\)
>     PATH=/home/soe/tomlinso/bin&#8203;:/product/cmiss/bin&#8203;:/usr/local/bin&#8203;:/opt/freeware/bin&#8203;:/usr/bin&#8203;:/etc&#8203;:/usr/sbin&#8203;:/usr/ucb&#8203;:/usr/vacpp/bin&#8203;:/usr/bin/X11&#8203;:/sbin&#8203;:/usr/java130/jre/bin&#8203;:/usr/java130/bin
>     PERL\_BADLANG \(unset\)
>     SHELL=/product/cmiss/bin/zsh

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Configure has changed so that it checks the size of pointers before the use64bitall script from hints/aix\.sh is run so

sh Configure \-de \-Duse64bitall \-Dusedevel

turns off the use64bitall making 64bitint instead\.  Perhaps aix\.sh needs to change $cc \(like IRIX\_6\.sh\) or set compiler flags to include \-q64\.

However\,

OBJECT\_MODE=64 sh Configure \-de \-Duse64bitall \-Dusedevel

uses the environment variable OBJECT\_MODE to affect cc so that it produces 64\-bit pointers\.
This produces a 64all perl that passes all tests\.
The integers no longer appear to be rounded in the comparison\.

Thank you\.

Jarkko Hietaniemi wrote&#8203;:
> 
> On Wed\, May 08\, 2002 at 06&#8203;:44&#8203;:06PM \+1200\, Karl Tomlinson wrote&#8203;:
> >
> > This is a bug report for perl from k\.tomlinson@&#8203;auckland\.ac\.nz\,
> > generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> >
> >
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> >
> > File&#8203;::Find\.pm and FindBin\.pm fail their tests when building 64bitall
> > due to an error in Cwd&#8203;:getcwd\.  Cwd&#8203;:abs\_path compares device numbers
> > \(which have the most significant 64th bit set on AIX\) and doesn't
> > detect a difference in the less significant bits\.  This results in an
> > invalid directory from getcwd&#8203;:
> 
> Could you try the latest 5\.8\.0\-to\-be developer snapshot&#8203;:
> 
>         http&#8203;://www\.iki\.fi/jhi/perl@&#8203;16493\.tgz
>         http&#8203;://www\.iki\.fi/jhi/perl@&#8203;16493\.tbz
> 
>         ftp&#8203;://ftp\.funet\.fi/pub/languages/perl/snap/perl@&#8203;16493\.tgz
>         ftp&#8203;://ftp\.funet\.fi/pub/languages/perl/snap/perl@&#8203;16493\.tbz
> 
>         ftp\.funet\.fi&#8203;::perlsnap
> 
> it has several fixes related to 64\-bitness\.
> 
> > % 32/perl \-we 'BEGIN\{@&#8203;INC="32/lib"\}; use Cwd; print getcwd\(\)\,"\\n"'
> > /home/soe/tomlinso/perl
> > % 64/perl \-we 'BEGIN\{@&#8203;INC="64/lib"\}; use Cwd; print getcwd\(\)\,"\\n"'
> > /soe/tomlinso/perl
> >
> > The incorrect directory is a result of abs\_path believing that /home
> > and / are the same directory&#8203;:
> >
> > % 64/perl \-we 'printf "%x %x\\n"\, \(stat "/home"\)\[0\.\.1\]'
> > 8000000a00000008 2
> > % 64/perl \-we 'printf "%x %x\\n"\, \(stat "/"\)\[0\.\.1\]'
> > 8000000a00000004 2
> > % 32/perl \-we 'printf "%x %x\\n"\, \(stat "/home"\)\[0\.\.1\]'
> > a0008 2
> > % 32/perl \-we 'printf "%x %x\\n"\, \(stat "/"\)\[0\.\.1\]'
> > a0004 2
> >
> > Simply including
> >
> > use integer
> >
> > in Cwd\.pm corrects this problem\.  I assume the device numbers were being
> > converted to doubles before comparison\.
> >
> > This works fine for 32 and 64\-bit perls on AIX\, but I don't know
> > whether this could cause problems for 32\-bit perls on a system with
> > 64\-bit device numbers\.  Would the \`use integer' truncate the device
> > number to 32\-bits?  Does that mean that using \`eq' and \`ne' instead of
> > \`==' and \`\!=' is a better solution?
> >
> > \[Please do not change anything below this line\]
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\- \-\-\-
> > Flags&#8203;:
> >     category=library
> >     severity=high
> > \-\-\-
> > Site configuration information for perl v5\.6\.1&#8203;:
> >
> > Configured by tomlinso at Wed May  8 17&#8203;:26&#8203;:18 NZST 2002\.
> >
> > Summary of my perl5 \(revision 5\.0 version 6 subversion 1\) configuration&#8203;:
> >   Platform&#8203;:
> >     osname=aix\, osvers=5\.1\.0\.0\, archname=aix\-64all
> >     uname='aix hpc1 1 5 0020abea4c00 '
> >     config\_args='\-de \-Dmksymlinks \-Dprefix=/product/cmiss \-Dbin=/product/cmiss/bin/64 \-Dsitebin=/product/cmiss/bin/64 \-Duse64bitall \-Ausenm=false'
> >     hint=recommended\, useposix=true\, d\_sigaction=define
> >     usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
> >     useperlio=undef d\_sfio=undef uselargefiles=define usesocks=undef
> >     use64bitint=define use64bitall=define uselongdouble=undef
> >   Compiler&#8203;:
> >     cc='cc'\, ccflags ='\-D\_ALL\_SOURCE \-D\_ANSI\_C\_SOURCE \-D\_POSIX\_SOURCE \-qmaxmem=16384 \-I/usr/local/include  \-D\_LARGE\_FILES \-qlonglong \-q64 \-DUSE\_64\_BIT\_ALL'\,
> >     optimize='\-O'\,
> >     cppflags='\-D\_ALL\_SOURCE \-D\_ANSI\_C\_SOURCE \-D\_POSIX\_SOURCE \-qmaxmem=16384 \-I/usr/local/include'
> >     ccversion='undef'\, gccversion=''\, gccosandvers=''
> >     intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=87654321
> >     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=8
> >     ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
> >     alignbytes=8\, usemymalloc=n\, prototype=define
> >   Linker and Libraries&#8203;:
> >     ld='ld'\, ldflags ='\-L/usr/local/lib  \-q64'
> >     libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib
> >     libs=\-lbind \-lnsl \-ldbm \-ldl \-lld \-lm \-lC \-lc \-lcrypt \-lbsd \-lPW \-liconv
> >     perllibs=\-lbind \-lnsl \-ldl \-lld \-lm \-lC \-lc \-lcrypt \-lbsd \-lPW \-liconv
> >     libc=\, so=a\, useshrplib=false\, libperl=libperl\.a
> >   Dynamic Linking&#8203;:
> >     dlsrc=dl\_aix\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='  \-bE&#8203;:/product/cmiss/lib/perl5/5\.6\.1/aix\-64all/CORE/perl\.exp'
> >     cccdlflags=' '\, lddlflags='\-b64 \-bhalt&#8203;:4 \-bM&#8203;:SRE \-bI&#8203;:$\(PERL\_INC\)/perl\.exp \-bE&#8203;:$\(BASEEXT\)\.exp \-b noentry \-lC \-lc \-L/usr/local/lib'
> >
> > Locally applied patches&#8203;:
> >
> >
> > \-\-\-
> > @&#8203;INC for perl v5\.6\.1&#8203;:
> >     /product/cmiss/lib/perl5/5\.6\.1/aix\-64all
> >     /product/cmiss/lib/perl5/5\.6\.1
> >     /product/cmiss/lib/perl5/site\_perl/5\.6\.1/aix\-64all
> >     /product/cmiss/lib/perl5/site\_perl/5\.6\.1
> >     /product/cmiss/lib/perl5/site\_perl
> >     \.
> >
> > \-\-\-
> > Environment for perl v5\.6\.1&#8203;:
> >     HOME=/home/soe/tomlinso
> >     LANG=en\_US
> >     LANGUAGE \(unset\)
> >     LC\_\_FASTMSG=true
> >     LD\_LIBRARY\_PATH \(unset\)
> >     LIBPATH \(unset\)
> >     LOGDIR \(unset\)
> >     PATH=/home/soe/tomlinso/bin&#8203;:/product/cmiss/bin&#8203;:/usr/local/bin&#8203;:/opt/freeware/bin&#8203;:/usr/bin&#8203;:/etc&#8203;:/usr/sbin&#8203;:/usr/ucb&#8203;:/usr/vacpp/bin&#8203;:/usr/bin/X11&#8203;:/sbin&#8203;:/usr/java130/jre/bin&#8203;:/usr/java130/bin
> >     PERL\_BADLANG \(unset\)
> >     SHELL=/product/cmiss/bin/zsh
> 
> \-\-
> $jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
>         \# There is this special biologist word we use for 'stable'\.
>         \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @jhi

On Thu\, May 09\, 2002 at 06&#8203;:00&#8203;:54PM \+1200\, Karl Tomlinson wrote&#8203;:
> Configure has changed so that it checks the size of pointers before the use64bitall script from hints/aix\.sh is run so
> 
> sh Configure \-de \-Duse64bitall \-Dusedevel
> 
> turns off the use64bitall making 64bitint instead\.

Ooops\, that's definitely a bug\.  Will fix for the next snapshot\.
\(Out tomorrow Friday\.\)

> Perhaps aix\.sh needs to change $cc \(like IRIX\_6\.sh\) or set compiler flags to include \-q64\.

I think we tried those many moons ago and they lead to quite a bit of trouble\.

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @tux

On Thu 09 May 2002 08&#8203;:00\, Karl Tomlinson \<k\.tomlinson@&#8203;auckland\.ac\.nz> wrote&#8203;:
> Configure has changed so that it checks the size of pointers before the use64bitall script from hints/aix\.sh is run so
> 
> sh Configure \-de \-Duse64bitall \-Dusedevel
> 
> turns off the use64bitall making 64bitint instead\.  Perhaps aix\.sh needs to change $cc \(like IRIX\_6\.sh\) or set compiler flags to include \-q64\.
> 
> However\,
> 
> OBJECT\_MODE=64 sh Configure \-de \-Duse64bitall \-Dusedevel
> 
> uses the environment variable OBJECT\_MODE to affect cc so that it produces 64\-bit pointers\.
> This produces a 64all perl that passes all tests\.
> The integers no longer appear to be rounded in the comparison\.
> 
> Thank you\.

Thanks for the fine analisys too\, might want to remember that for future hint
changes\.

\-\- 
H\.Merijn Brand        Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using perl\-5\.6\.1\, 5\.7\.3 & 631 on HP\-UX 10\.20 & 11\.00\, AIX 4\.2\, AIX 4\.3\,
&nbsp;  WinNT 4\, Win2K pro & WinCE 2\.11\.  Smoking perl CORE&#8203;: smokers@&#8203;perl\.org
http&#8203;://archives\.develooper\.com/daily\-build@&#8203;perl\.org/   perl\-qa@&#8203;perl\.org
send smoke reports to&#8203;: smokers\-reports@&#8203;perl\.org\, QA&#8203;: http&#8203;://qa\.perl\.org

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

This is fixed in 5\.8\.0\-RC1\.

Thanks to those who fixed it\.

perlbug@&#8203;onion\.perl\.org wrote&#8203;:

>                 This is a Perlbug status\(open\) reminder for an outstanding
>                 bug\, a report for which is appended at the base of this email\.
>
>                 If the the status of this bug is in any way incorrect\, please
>                 inform an administrator of the Perlbug system\.
>
>                 Further data relating to this bug\(20020507\.021\) may be found at&#8203;:
>
>                         http&#8203;://bugs\.perl\.org/perlbug\.cgi?req=bidmid&bidmid=20020507\.021
>
>                 The group\(library\) of administrators responsible for this bug is&#8203;:
>
>                         http&#8203;://bugs\.perl\.org/perlbug\.cgi?req=group\_id&group\_id=2
>
>                 For email help send an email to&#8203;:
>
>                         To&#8203;: bugdb@&#8203;perl\.org
>                         Subject&#8203;: \-H
>
>                 Bug report \(current status\) follows&#8203;:
>
>
> Bug&#8203;:      20020507\.021
> Subject&#8203;:  64\-bit getcwd error due to rounding of device number
> From&#8203;:     Karl Tomlinson \<k\.tomlinson@&#8203;auckland\.ac\.nz>
> Created&#8203;:  2002\-05\-07 23&#8203;:44&#8203;:22  Modified&#8203;: 2002\-05\-07 23&#8203;:44&#8203;:22
>
> Status&#8203;:   open             Severity&#8203;: high
> OS names&#8203;: aix
> Group&#8203;:    library
> Version&#8203;:  5\.6\.1\, 5\.061     Fixed in&#8203;:
>
> Message ids&#8203;: 4
> Patch ids&#8203;:   0
> Test ids&#8203;:    0
> Note ids&#8203;:    0
> Change ids&#8203;:  0
> Parent ids&#8203;:  0
> Child ids&#8203;:   0
> Address ids&#8203;: 0
>
>

**Comment by p5pRT** [comment]
### From @jhi

On Tue\, Jul 09\, 2002 at 01&#8203;:37&#8203;:20AM \+1200\, Karl Tomlinson wrote&#8203;:
> This is fixed in 5\.8\.0\-RC1\.
> 
> Thanks to those who fixed it\.

It was fixed on 2002\-05\-09 based on your good analysis of the problem\.

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen
